### PR TITLE
Refactor PKCT01

### PR DESCRIPTION
**Pull request by edelarua**
Blocked by https://github.com/insightsengineering/tern/pull/1048.

### CODE REVIEW

**Comment by shajoezhu** [comment]
hi @qiliu1013 , can you please help to review this template snapshot. Thanks!

https://github.com/insightsengineering/tlg-catalog/blob/f6d52f02de44de5c34351ac54d9da8bd6ba0637f/package/tests/testthat/_snaps/pkct01/markdown-snaps.md

**Review by Melkiades — COMMENTED** [review]
There is an error that looks unrelated:
```
  Quitting from lines 45-118 [variant1] (adat03.qmd)
  Error:
  ! Error in content (summary) function: unused argument (cache_env = <environment>)
  	occured at path: ARM[A: Drug X]
```

File: book/tables/pharmacokinetic/pkct01.qmd [hunk 1]
```diff
@@ -26,78 +26,116 @@ adpc_1 <- adpc %>%
     AVALCAT1 = as.factor(AVALCAT1)
   ) %>%
   filter(ACTARM == "A: Drug X", PARAM == "Plasma Drug X") %>%
-  mutate(ACTARM = factor(ACTARM, levels = c("A: Drug X"))) %>%
-  select(NFRLT, ACTARM, VISIT, AVAL, PARAM, AVALCAT1)
+  select(NFRLT, ACTARM, VISIT, AVAL, PARAM, AVALCAT1) %>%
+  var_relabel(NFRLT = "Nominal Time from First Dose (hr)")
```
Comment (Melkiades) [inline book/tables/pharmacokinetic/pkct01.qmd:30]: is this a typo in rcd?

File: package/tests/testthat/_snaps/pkct01/markdown-snaps.md [hunk 2]
```diff
@@ -8,27 +8,96 @@
...
+      ND: Not Derived
+
+# result_v3
+
+    Code
+      print(data_snap[[i]])
+    Output
+      Summary of PK Concentrations by Nominal Time and Treatment: PK Evaluable
+      Protocol: xxxxx
+      Analyte:  Plasma Drug X
```
Comment (Melkiades) [inline package/tests/testthat/_snaps/pkct01/markdown-snaps.md:77]: @qiliu1013 can you check if this is correct (imputation rule 1/2)

@shajoezhu fyi

File: package/tests/testthat/_snaps/pkct01/markdown-snaps.md [hunk 3]
```diff
@@ -8,27 +8,96 @@
...
 
+# result_v2
+
+    Code
+      print(data_snap[[i]])
+    Output
+      Summary of PK Concentrations by Nominal Time and Treatment: PK Evaluable
+      Protocol: xxxxx
+      Analyte:  Plasma Drug X
+      Treatment: A: Drug X
```
Comment (Melkiades) [inline package/tests/testthat/_snaps/pkct01/markdown-snaps.md:43]: @qiliu1013 can you check if this is correct (imputation rule 1/3)

@shajoezhu fyi

**Review by Melkiades — APPROVED** [review]
Lgtm! Thanks Emily! It is very good :)

File: book/tables/pharmacokinetic/pkct01.qmd (hunk 1 above)
Comment (edelarua) [reply book/tables/pharmacokinetic/pkct01.qmd:30]: No, just in the template - a correct (slightly different) label is used in rcd.

File: package/tests/testthat/_snaps/pkct01/markdown-snaps.md (hunk 3 above)
Comment (qiliu1013) [reply package/tests/testthat/_snaps/pkct01/markdown-snaps.md:43]: Looks good! Thanks.

File: package/tests/testthat/_snaps/pkct01/markdown-snaps.md (hunk 2 above)
Comment (qiliu1013) [reply package/tests/testthat/_snaps/pkct01/markdown-snaps.md:77]: Looks good! Thanks.

File: package/tests/testthat/_snaps/pkct01/markdown-snaps.md [hunk 4]
```diff
@@ -8,27 +8,96 @@
...
+          Nominal Time from First Dose (hr)          Number                                                                                                         
+                                                       of                                                                                                           
+                                               n    LTRs/BLQs    Mean       SD     CV (%) Mean   Geometric Mean   CV % Geometric Mean   Median    Minimum    Maximum
+      ——————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
+      A: Drug X (N=134)                                                                                                                                             
+        Day 1                                                                                                                                                       
+          0                                   134      134       0        0            NE            0                    NE             0         0          0
```
Comment (anajens) [inline package/tests/testthat/_snaps/pkct01/markdown-snaps.md:20]: @edelarua not sure that the 0 should display here for geometric mean?

Per GDSR:
image

**Review by anajens — APPROVED** [review]
lgtm 👍

**Pull request merged**